### ISSUE TITLE
Issue: tf.function not working when dealing with tf.stack

### ISSUE THREAD

**Issue by giostefa** [tensorflow/tensorflow#48278]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Windows 10
- TensorFlow version 2.3.0

**Describe the current behavior**
tf.function is not working in converting functions containing tf.stack.

**Describe the expected behavior**
I would expect tf.function to actually manage to convert the function

**Standalone code to reproduce the issue**
```
import tensorflow as tf

c = tf.Variable([[1., 5.], [2., 4.]])

@tf.function
def toy_fct(x):
    y = tf.stack([x[0,:], x[1,:]], axis=0)
    return y

toy_fct(c)
```

**Other info / logs**
The warning given is:

> WARNING:tensorflow:AutoGraph could not transform <function toy_fct at 0x000001DC9A13A670> and will run it as-is.
Please report this to the TensorFlow team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output.
Cause: module 'gast' has no attribute 'Index'
To silence this warning, decorate the function with @tf.autograph.experimental.do_not_convert
WARNING: AutoGraph could not transform <function toy_fct at 0x000001DC9A13A670> and will run it as-is.
Please report this to the TensorFlow team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output.
Cause: module 'gast' has no attribute 'Index'
To silence this warning, decorate the function with @tf.autograph.experimental.do_not_convert
<tf.Tensor: shape=(2, 2), dtype=float32, numpy=
array([[1., 5.],
       [2., 4.]], dtype=float32)>

**Comment by giostefa** [comment]
The issue seems much broader as the real problem is tf.function fails to deal with tensor slicing. Closing the issue since there are many more out there on the same subject.

**Comment by ymodak** [comment]
Similar issue was recently [fixed]( https://github.com/tensorflow/tensorflow/issues/48289#issuecomment-820789715) and should be available with upcoming TF 2.5.0-rc2 release.
